pkg/kubelet/prober/worker.go

```diff
@@ -265,8 +265,11 @@ func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 		return false
 	}
 
-	// Probe disabled for InitialDelaySeconds.
-	if int32(time.Since(c.State.Running.StartedAt.Time).Seconds()) < w.spec.InitialDelaySeconds {
+	// Probe disabled for InitialDelaySeconds only if container's StartedAt time is
+	// earlier than system current time. But probe must be on for time drift case,
+	// This may happen due to nvram flash which resets system BIOS settings and RTC.
+	startedAtInSecs := time.Since(c.State.Running.StartedAt.Time).Seconds()
+	if startedAtInSecs > 0 && int32(startedAtInSecs) < w.spec.InitialDelaySeconds {
```

Comment:
Would it make sense to wait `InitialDelaySeconds` since `doProbe` called first time if `startedAtInSecs <= 0`.

Reply:
@bart0sh good point, but would that happen (i.e. pod startup followed by immediate time drift event before `InitialDelaySeconds`) ? In our case, it just happened during node maintenance.

Reply:
My point was that if `startedAtInSecs <= 0` probe would run immediately, which is not desirable. It's better to wait for InitialDelaySeconds just to be on the safe side.

Reply:
ok @bart0sh , added code to wait for `InitialDelaySeconds` when time drift happens.